### PR TITLE
refactor EndPointForcesSinusoidal example and test cases

### LINKED ISSUE #103: Refactor `EndpointForcesSinusoidal` used by joint examples
We can move `EndpointForcesSinusoidal` inside the `elastica.external_forces` since there are test cases already defined for this class. 

https://github.com/GazzolaLab/PyElastica/blob/9e461aa061101fe94af245bc9219730e30b4a9d7/examples/JointCases/external_force_class_for_joint_test.py#L6

### PR DESCRIPTION
**Pull request by armantekinalp**
Fixes #103 . This fix is required for implementing pyproject for PyElastica.

### CODE REVIEW

File: elastica/external_forces.py [hunk 1]
```diff
@@ -470,3 +471,73 @@ def inplace_substraction(external_force_or_torque, force_or_torque):
...
+        normal_direction=np.array([0, 1, 0]),
+    ):
+        super(EndpointForcesSinusoidal, self).__init__()
+        # Start force
+        self.start_force_mag = start_force_mag
+        self.end_force_mag = end_force_mag
+
+        # Applied force directions
+        self.normal_direction = normal_direction
+        # self.roll_direction = np.cross(tangent_direction, normal_direction)
```
Comment (bhosale2) [inline elastica/external_forces.py:495]: remove?

File: elastica/external_forces.py [hunk 2]
```diff
@@ -470,3 +471,73 @@ def inplace_substraction(external_force_or_torque, force_or_torque):
     for i in range(3):
         for k in range(blocksize):
             external_force_or_torque[i, k] -= force_or_torque[i, k]
+
+
+# Force class. This force class is used only for joint test cases
+class EndpointForcesSinusoidal(NoForces):
+    """Applies sinusoidal forces on endpoints"""
```
Comment (bhosale2) [inline elastica/external_forces.py:478]: maybe docstring here better reflects the comments you have below?

**Review by bhosale2 — CHANGES_REQUESTED** [review]
LGTM mostly, 2 minor requests.

**Review by skim0119 — CHANGES_REQUESTED** [review]
Can you add the `EndpointForcesSinusoidal` on documentation?

File: elastica/external_forces.py (hunk 1 above)
Comment (armantekinalp) [reply elastica/external_forces.py]: Done.

**Comment by armantekinalp** [comment]
I added documentation. Could you check again and then we can merge.

**Comment by skim0119** [comment]
here: https://github.com/armantekinalp/PyElastica/blob/103_refactor_EndpointForcesSinusoidal/docs/api/external_forces.rst

**Review by bhosale2 — APPROVED** [review]
LGTM.

**Review by skim0119 — APPROVED** [review]
LGTM

**Pull request merged**